### PR TITLE
ci: Build on Go 1.18, bump actions versions

### PR DESCRIPTION
**Pull request by francislavoie**

### CODE REVIEW

**Comment by mohammed90** [comment]
We should stick to go1.17 until [github.com/lucas-clemente/quic-go](https://github.com/lucas-clemente/quic-go) is compatible with go1.18.

**Review by mohammed90 — APPROVED** [review]
Thanks, Francis! Let's merge 🔥

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ ubuntu-latest, macos-latest, windows-latest ]
-        go: [ '1.16', '1.17' ]
+        go: [ '1.16', '1.17', '1.18' ]
```
Comment (mholt) [inline .github/workflows/ci.yml:22]: We can remove the tests for Go 1.16. We only support the most recent 2 releases of Go.

File: duration_fuzz.go [hunk 2]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//go:build gofuzz
 // +build gofuzz
```
Comment (mholt) [inline duration_fuzz.go:16]: What's with both of these? Seems redundant...

File: .github/workflows/ci.yml (hunk 1 above)
Comment (francislavoie) [reply .github/workflows/ci.yml:22]: We're keeping 1.16 minimum until COPR can use 1.17 on all their supported platforms

File: duration_fuzz.go (hunk 2 above)
Comment (francislavoie) [reply duration_fuzz.go:16]: See the 1.18 release notes, they're migrating from one syntax to the other, keeping both for now for compat. This is from running gofmt, I didn't add these by hand

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mholt) [reply .github/workflows/ci.yml:22]: These are just CI tests though, not releases, right? I don't think we need to triple our CI tests for COPRs. (I really do not like the idea of external distribution platforms holding us back. The added CI tests do slow things down.)

File: duration_fuzz.go (hunk 2 above)
Comment (mholt) [reply duration_fuzz.go:16]: Oh! TIL. Very cool.

I installed Go 1.18 today but haven't done much with it yet.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (francislavoie) [reply .github/workflows/ci.yml:22]: It doesn't slow things down, the tests are run in parallel.

**Pull request merged**